Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5774,41 +5775,38 @@ static bool hasVisibleBoxDecorationsOrBackground(const RenderElement& renderer)
     return renderer.hasVisibleBoxDecorations() || renderer.style().hasOutline();
 }
 
-// Constrain the depth and breadth of the search for performance.
-static const unsigned maxRendererTraversalCount = 200;
-
-static void determineNonLayerDescendantsPaintedContent(const RenderElement& renderer, unsigned& renderersTraversed, RenderLayer::PaintedContentRequest& request)
-{
 #if HAVE(HDR_SUPPORT)
-    auto hasPaintedImageHDRContent = [] (const RenderObject& childElement) {
+static bool drawsHDRContent(const RenderElement& renderer)
+{
 #if ENABLE(HDR_FOR_IMAGES)
-        CheckedPtr imageRenderer = dynamicDowncast<RenderImage>(childElement);
-        if (!imageRenderer)
-            return false;
-
-        if (auto* cachedImage = imageRenderer->cachedImage())
-            return cachedImage->drawsHDRContent();
-#else
-        UNUSED_PARAM(childElement);
-#endif
+    if (CheckedPtr imageRenderer = dynamicDowncast<RenderImage>(renderer)) {
+        if (auto* cachedImage = imageRenderer->cachedImage()) {
+            if (cachedImage->drawsHDRContent())
```

Comment:
It's odd to an an image if it "draws HDR content"; the right question would be "isHDR"? Same with canvas.

Reply:
This code was moved to a separate function named `isReplacedElementWithHDR()`. The `CachedImage::drawsHDRContent()` was renamed `CachedImage::isHDR()`.